### DISCUSSION TITLE
error: [handleAbortError] AI response error; aborting request: Config not found for the ollama custom endpoint.

### DISCUSSION THREAD

**Discussion by lexciobotariu** [danny-avila/LibreChat#4677] [Troubleshooting] [answered]
### What happened?

I'm getting this error when I run the app with ollama separately. 

 This is how my librechat.yaml looks like for the ollama model

```    
endpoints:
  custom:
    - name: "Ollama"
      apiKey: "ollama"
      # use 'host.docker.internal' instead of localhost if running LibreChat in a docker container
      baseURL: "http://127.0.0.1:11434/v1/chat/completions" 
      models:
        default: [
          "llama2",
          "mistral",
          "codellama",
          "dolphin-mixtral",
          "mistral-openorca"
          ]
        # fetching list of models is supported but the `name` field must start
        # with `ollama` (case-insensitive), as it does in this example.
        fetch: true
      titleConvo: true
      titleModel: "current_model"
      summarize: false
      summaryModel: "current_model"
      forcePrompt: true
      modelDisplayLabel: "Ollama"
      addParams:
          "stop": [
              "<|start_header_id|>",
              "<|end_header_id|>",
              "<|eot_id|>",
              "<|reserved_special_token"
          ]
```

When I access the chat it finds the model I have installed, but when I type a message I get this error 
```
Something went wrong. Here's the specific error message we encountered: An error occurred while processing your request. Please contact the Admin.
```
Also I see in the vscode console `warn: Invalid conversation ID: null` I added the whole error down below. 

I checked the indentation of the YAML file, and it seems correct. 

### Steps to Reproduce

1. Install Ollama, download any model
2. Configure librachat.yaml as above.
3. Run the project with NPM

### Relevant log output

```shell
2024-11-08 16:55:53 error: [handleAbortError] AI response error; aborting request: Config not found for the ollama custom endpoint.
2024-11-08 16:55:53 warn: Invalid conversation ID: null
2024-11-08 16:55:53 info: ---`saveMessage` context: undefined
2024-11-08 16:55:53 info: ---Invalid conversation ID Params: {
  "sender": "Ollama",
  "messageId": "aa6c5bcf-a813-46a6-a37b-db60c7f000eb",
  "conversationId": null,
  "parentMessageId": "00000000-0000-0000-0000-000000000000",
  "unfinished": false,
  "error": true,
  "final": true,
  "text": "An error occurred while processing your request. Please contact the Admin.",
  "isCreatedByUser": false,
  "user": "672df0d453e07c7ddd08eafa"
}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by m4druga** [comment]
I'm dealing with the same issue.

**Comment by danny-avila** [comment]
Try updating to the latest build, this issue was resolved

**Comment by m4druga** [comment, marked as answer]
I didn't find the root cause, but I found a way to work around the problem: the value of name needs to be in lowercase.

endpoints:
  custom:
    - name: "ollama"
      apiKey: "ollama"

This is new, as previous deployments used name with only the first letter in uppercase without causing this error. Interestingly, the example on the [LibreChat documentation site](https://www.librechat.ai/docs/configuration/librechat_yaml/ai_endpoints/ollama) still shows name with an uppercase 'O'.

**Comment by daSonne** [comment]
this is driving crazy 
running liberchat + ollama on docker (mac)

(librechat.yaml)
custom:
    - name: "ollama"
      apiKey: "ollama"
      # use 'host.docker.internal' instead of localhost if running LibreChat in a docker container
      baseURL: "http://host.docker.internal:11434/v1/" 
      models:
        default: [
          "llama2",
          "mistral",
          "codellama",
          "dolphin-mixtral",
          "mistral-openorca"
          ]
        # fetching list of models is supported but the `name` field must start
        # with `ollama` (case-insensitive), as it does in this example.
        fetch: true
      titleConvo: true
      titleModel: "current_model"
      summarize: false
      summaryModel: "current_model"
      forcePrompt: false
      modelDisplayLabel: "Ollama"
I tried every solution I found no luck !
help!

**Comment by himefei** [comment]
Hi guys, for those who still can't get it to work, and you have already tried updating, change Ollama to lowercase. Please try https://github.com/danny-avila/LibreChat/discussions/2745, it worked out for me by disableing Meli search

**Comment by danny-avila** [comment]
Apologies, it seems I missed something in my update, it should be fixed after this is merged and to remain case-insensitive:

https://github.com/danny-avila/LibreChat/pull/4681
